pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -634,9 +634,12 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	controller.AddCephVersionLabelToDeployment(c.clusterInfo.CephVersion, deployment)
 	controller.AddCephVersionLabelToDeployment(c.clusterInfo.CephVersion, deployment)
 	k8sutil.SetOwnerRef(&deployment.ObjectMeta, &c.clusterInfo.OwnerRef)
+
 	p := cephv1.GetOSDPlacement(c.spec.Placement)
```

Comment:
Unrelated to this PR, seems like we could simplify the pattern to be:
```go
p := cephv1.GetOSDPlacement(c.spec.Placement)
if osdProps.onPVC() {
  osdProps.getPreparePlacement().ApplyToPodSpec(&podSpec)
}
p.ApplyToPodSpec(&podSpec) 
```
no need to duplicate the last line in both the if and else if